cluster/node.go

```diff
@@ -440,10 +435,18 @@ func (n *Node) String() string {
 
 func (n *Node) handler(ev *dockerclient.Event, _ chan error, args ...interface{}) {
 	// Something changed - refresh our internal state.
-	if ev.Status == "pull" || ev.Status == "untag" || ev.Status == "delete" {
+	switch ev.Status {
+	case "pull", "untag", "delete":
+		// These events refer to images so there's no need to update
+		// containers.
 		n.refreshImages()
-	} else {
-		n.refreshContainer(ev.Id)
+	case "start", "die":
+		// If the container is started or stopped, we have to do an inspect in
```

Comment:
`stop` is always preceded by `die` as far as I can tell, so this is to avoid doing inspect twice when stopping.